frontend/views/containers/proposals/RemoveMember.vue

```diff
@@ -1,58 +1,155 @@
+    avatar.c-avatar(:src='memberGlobalProfile.picture' size='lg')
+    i18n.is-title-4(tag='p' :args='{ name: userDisplayName(username) }' v-if='groupShouldPropose') Remove {name} from your group
+    i18n.is-title-4(tag='p' :args='{ name: userDisplayName(username) }' v-else) Are you sure you want to remove {name} from your group?
+
+  banner-scoped(ref='formMsg' data-test='proposalError')
 </template>
 
 <script>
+import { mapState, mapGetters } from 'vuex'
+import sbp from '~/shared/sbp.js'
+import { CLOSE_MODAL } from '@utils/events.js'
+import L from '@view-utils/translations.js'
 import Avatar from '@components/Avatar.vue'
+import { PROPOSAL_REMOVE_MEMBER, STATUS_OPEN } from '@model/contracts/voting/proposals.js'
+import BannerScoped from '@components/banners/BannerScoped.vue'
 import ProposalTemplate from './ProposalTemplate.vue'
+import { validateRemoveMember } from '@model/contracts/group.js'
 
 export default {
   name: 'RemoveMember',
   components: {
     Avatar,
+    BannerScoped,
     ProposalTemplate
   },
+  props: {
+    username: String
+  },
   data () {
     return {
       ephemeral: {
         currentStep: 0
       },
       config: {
         steps: [
-          'AddMemberRule'
+          'RemoveMember'
         ]
       }
     }
   },
   created () {
-    console.log('TODO - Get username info')
+    if (!this.username) {
+      console.warn('Missing username to display RemoveMember modal')
+      sbp('okTurtles.events/emit', CLOSE_MODAL)
+    }
   },
   computed: {
-    member () {
-      // TODO - connect to real store.
-      return {
-        id: 1000,
-        displayName: '[Attila the Hun]',
-        picture: '/assets/images/default-avatar.png'
-      }
+    ...mapState([
+      'currentGroupId'
+    ]),
+    ...mapGetters([
+      'currentGroupState',
+      'globalProfile',
+      'groupProfiles',
+      'groupSettings',
+      'groupShouldPropose',
+      'groupMembersCount',
+      'ourUsername',
+      'userDisplayName'
+    ]),
+    memberGlobalProfile () {
+      return this.globalProfile(this.username) || {}
+    },
+    rule () {
+      const { threshold } = this.groupSettings.proposals['remove-member'].ruleSettings.threshold
+      return { value: Math.round(this.groupMembersCount * threshold), total: this.groupMembersCount }
     }
   },
   methods: {
-    submit (form) {
-      console.log(
-        'TODO: Logic to Propose removing a member.',
-        'member:', 'DUMMY',
-        'reason:', form.reason
-      )
+    validateRemoval (data) {
+      return validateRemoveMember(this.currentGroupState, {
+        data,
+        meta: { username: this.ourUsername }
+      })
+    },
+    async submit (form) {
+      this.$refs.formMsg.clean()
+      const member = this.username
+      const memberID = this.groupProfiles[member].contractID
+
+      if (this.groupShouldPropose) {
+        // Look for existing proposal
+        for (const hash in this.currentGroupState.proposals) {
```

Comment:
This is similar to the validation we "have" (I commented it) on `contract/group.js` but specific for removeMember proposal. Should we do it there too?